### ISSUE TITLE
Android Apollo (GraphQL) autogeneration issue with Kotlin

### ISSUE THREAD

**Issue by sneronovAdviqo** [apollographql/apollo-android#1953]
I used to use an Android library Apollo (GraphQL) version 1.2 with Java models autogeneration. Today I updated to version 1.3 with possibility to generate Kotlin (model) data classes. But I have got an issue. In the Schema I have an Integer parameter:

```
...
{
   "name": "size",
   "description": "",
   "type": {
      "kind": "SCALAR",
      "name": "Int",
      "ofType": null
   },
   "defaultValue": null
}
...
```
Apollo generates code, where it uses Kotlin synthetic sugar by lambda statement "apply{}" on List parameter. Inside it applies to a property of generated class "size", but inside "apply{}" the "size" is the property of the "apply{}" context, not of the generated class. And, of course, "size" does not have "size.defined" and "size.value":

```
data class MembersQuery(
  val mandatorNo: Int,
  val memberNoFilter: Input<List<Int?>> = Input.absent(),
  val memberId: Input<String> = Input.absent(),
  val page: Int,
  val size: Input<Int> = Input.absent()
) : Query<MembersQuery.Data, MembersQuery.Data, Operation.Variables> {
  @Transient
  private val variables: Operation.Variables = object : Operation.Variables() {
    override fun valueMap(): Map<String, Any?> = mutableMapOf<String, Any?>().apply {
      this["mandatorNo"] = mandatorNo
      if (memberNoFilter.defined) this["memberNoFilter"] = memberNoFilter.value
      if (memberId.defined) this["memberId"] = memberId.value
      this["page"] = page
      if (size.defined) this["size"] = size.value
    }
...
}
```
Roll back to java generation is not good idea as I have changed many things, expecting of the Kotlin magic. Change the generated class does not have sense, as it will be regenerated and rewritten. What to do in the case?

**Comment by sneronovAdviqo** [comment]
Colleagues, when and in which version will be the fix released?

**Comment by martinbonnin** [comment]
@sneronovAdviqo that'll be fixed in the next version (1.3.1). It'll hopefully be released soon. You can watch the repo releases to be notified when this happens.

**Comment by sneronovAdviqo** [comment]
On version 1.3.2 the same problem!
